### PR TITLE
added Python 3.12 in the CI

### PR DESCRIPTION
**Pull request by auvipy**

### CODE REVIEW

**Comment by auvipy** [comment]
=================================== FAILURES ===================================
______________________ test_Hub.test_call_soon_uses_lock _______________________

self = <t.unit.asynchronous.test_hub.test_Hub object at 0x7fc18972a150>

    def test_call_soon_uses_lock(self):
        callback = Mock(name='callback')
        with patch.object(self.hub, '_ready_lock', autospec=True) as lock:
            self.hub.call_soon(callback)
>           assert lock.__enter__.called_once()

t/unit/asynchronous/test_hub.py:196: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='_ready_lock.__enter__' id='140469018516880'>
name = 'called_once'

    def __getattr__(self, name):
        if name in {'_mock_methods', '_mock_unsafe'}:
            raise AttributeError(name)
        elif self._mock_methods is not None:
            if name not in self._mock_methods or name in _all_magics:
                raise AttributeError("Mock object has no attribute %r" % name)
        elif _is_magic(name):
            raise AttributeError(name)
        if not self._mock_unsafe and (not self._mock_methods or name not in self._mock_methods):
            if name.startswith(('assert', 'assret', 'asert', 'aseert', 'assrt')) or name in _ATTRIB_DENY_LIST:
>               raise AttributeError(
                    f"{name!r} is not a valid assertion. Use a spec "
                    f"for the mock if {name!r} is meant to be an attribute.")
E               AttributeError: 'called_once' is not a valid assertion. Use a spec for the mock if 'called_once' is meant to be an attribute.

**Comment by ThiefMaster** [comment]
`assert somemock.called_once` does not exist and was always broken, you need to call `somemock.assert_called_once()` instead

**Comment by auvipy** [comment]
thanks Adrian

File: tox.ini [hunk 1]
```diff
@@ -131,4 +133,4 @@ commands =
     pydocstyle {toxinidir}/kombu
 
 [testenv:mypy]
-commands = python -m mypy --config-file setup.cfg
+commands = python -m mypy --config-file setup.cfg
```
Comment (ThiefMaster) [inline tox.ini:136]: you may want to configure your editor to automatically end a file with a linebreak ;)

**Comment by auvipy** [comment]
I think the failure should be fixed by https://github.com/celery/kombu/commit/3ad075a536d177170a7a5d7b04d81e94bd5bf404

**Comment by auvipy** [comment]
I think we got another similar change to do =================================== FAILURES ===================================
______________________ test_Hub.test__pop_ready_uses_lock ______________________

self = <t.unit.asynchronous.test_hub.test_Hub object at 0x7f6aacef6f30>

    def test__pop_ready_uses_lock(self):
        with patch.object(self.hub, '_ready_lock', autospec=True) as lock:
            self.hub._pop_ready()
>           assert lock.__enter__.called_once()

t/unit/asynchronous/test_hub.py:571: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='_ready_lock.__enter__' id='140096041911408'>
name = 'called_once'

    def __getattr__(self, name):
        if name in {'_mock_methods', '_mock_unsafe'}:
            raise AttributeError(name)
        elif self._mock_methods is not None:
            if name not in self._mock_methods or name in _all_magics:
                raise AttributeError("Mock object has no attribute %r" % name)
        elif _is_magic(name):
            raise AttributeError(name)
        if not self._mock_unsafe and (not self._mock_methods or name not in self._mock_methods):
            if name.startswith(('assert', 'assret', 'asert', 'aseert', 'assrt')) or name in _ATTRIB_DENY_LIST:
>               raise AttributeError(
                    f"{name!r} is not a valid assertion. Use a spec "
                    f"for the mock if {name!r} is meant to be an attribute.")
E               AttributeError: 'called_once' is not a valid assertion. Use a spec for the mock if 'called_once' is meant to be an attribute.

**Pull request merged**